src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -0,0 +1,363 @@
+        final List<InventoryEntryDraft> drafts = transformToDrafts(inventories);
+        processDrafts(drafts);
+    }
+
+    @Override
+    @Nonnull
+    public InventorySyncStatistics getStatistics() {
+        return statistics;
+    }
+
+    /**
+     * Instantiate and fill {@code supplyChannelKeyToId} with values fetched from supply channels in CTP.
+     * If {@code ensureChannels} flag from {@code syncOptions} is set to true, method checks if {@code drafts} reference
+     * any supply channels that are missing in CTP. If such channels exist, method attempt to create missing supply
+     * channels and update {@code supplyChannelKeyToId} with newly created values.
+     *
+     * @param drafts {@link List} of {@link InventoryEntryDraft} containing data that would be synced into CTP project.
+     */
+    private void buildChannelMap(@Nonnull final List<InventoryEntryDraft> drafts) {
+        supplyChannelKeyToId = inventoryService.fetchAllSupplyChannels()
+                .stream()
+                .collect(toMap(Channel::getKey, Channel::getId));
+        if (syncOptions.isEnsureChannels()) {
+            final List<String> missingChannelsKeys = drafts.stream()
+                    .map(SkuKeyTuple::of)
+                    .map(SkuKeyTuple::getKey)
+                    .distinct()
+                    .filter(key -> key != null)
+                    .filter(key -> !supplyChannelKeyToId.containsKey(key))
+                    .collect(toList());
+            missingChannelsKeys.stream()
+                    .map(this::createMissingSupplyChannel)
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .forEach(channel -> supplyChannelKeyToId.put(channel.getKey(), channel.getId()));
+        }
+    }
+
+
+    /**
+     * Method process batch of {@link InventoryEntryDraft}. The processing means:
+     * <ol>
+     *     <li>Fetching existing {@link InventoryEntry} that correspond to passed {@code drafts}</li>
+     *     <li>Deciding if each draft should be used for creating new entry or updating existing one</li>
+     *     <li>Attempting to create new entries or update existing ones</li>
+     *     <li>Updates the {@code statistics} object backed for this instance</li>
+     * </ol>
+     * @param drafts batch of drafts that need to be synced
+     */
+    private void processBatch(final List<InventoryEntryDraft> drafts) {
+        final Map<SkuKeyTuple, InventoryEntry> existingInventories = fetchExistingInventories(drafts);
+        drafts.forEach(draft -> {
+            final SkuKeyTuple skuKeyOfDraft = SkuKeyTuple.of(draft);
+            if (existingInventories.containsKey(skuKeyOfDraft)) {
+                final InventoryEntry existingEntry = existingInventories.get(skuKeyOfDraft);
+                attemptUpdate(existingEntry, draft);
+            } else {
+                attemptCreate(draft);
+            }
+            statistics.incrementProcessed();
```

Comment:
Yes, there exist inconsistency. Actually I point it out in doc related PR :) https://github.com/commercetools/commercetools-sync-java/pull/32#discussion_r115950504 There's still no agreement to what "processed" means.